File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution(object):
    def productExceptSelf(self, nums):
        prod, zero_counter = 1, 0
        for num in nums:
            if num != 0:
                prod *= num
            else:
                zero_counter += 1
        if zero_counter > 1:
            return [0] * len(nums)
        for i in range(len(nums)):
            if nums[i] != 0:
                if zero_counter == 1:
                    nums[i] = 0
                else:
                    nums[i] = prod // nums[i]
            else:
                nums[i] = prod
        return nums
        

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int prod = 1;
        int zero_counter = 0;
        for (int num : nums) {
            if (num != 0) {
                prod *= num;
            } else {
                zero_counter++;
            }
        }
        if (zero_counter > 1) {
            return new int[nums.length];
        } 
        for (int i = 0; i < nums.length; i++) {
            if (zero_counter == 1) {
                if (nums[i] == 0) {
                    nums[i] = prod;
                } else {
                    nums[i] = 0;
                } 
            } else {
                nums[i] = prod / nums[i];
            }
        }
        return nums;
    }
}